### PR TITLE
Apply fmt, fix clippy warnings and add them to CI

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -121,11 +116,15 @@ where
             inner: MerkleTree::<Db, M>::with_hasher(hasher),
         }
     }
-    pub fn push_leaf(&mut self, raw_data: &[u8], namespace: NamespaceId) -> Result<(), ()> {
+    pub fn push_leaf(
```
Comment (oblique) [inline src/lib.rs:120]: add an empty line above this

File: src/lib.rs [hunk 2]
```diff
@@ -353,7 +352,7 @@ where
                     .collect();
                 if let RangeProofType::Partial = self.check_range_proof(
                     root,
-                    &mut &leaf_hashes[..],
+                    &leaf_hashes[..],
```
Comment (oblique) [inline src/lib.rs]: `[..]` can be avoided

File: src/lib.rs [hunk 3]
```diff
@@ -409,12 +419,8 @@ mod tests {
...
                 let leaf_hashes: Vec<NamespacedHash> =
                     tree.leaves()[j..i].iter().map(|l| l.hash.clone()).collect();
-                let res = tree.check_range_proof(
-                    &root,
-                    &mut &leaf_hashes[..],
-                    &mut proof.take_siblings(),
-                    j,
-                );
+                let res =
+                    tree.check_range_proof(&root, &leaf_hashes[..], &mut proof.take_siblings(), j);
```
Comment (oblique) [inline src/lib.rs]: `[..]` can be avoided

File: src/nmt_proof.rs [hunk 4]
```diff
@@ -70,7 +67,7 @@ impl<M: NamespaceMerkleHasher<Output = NamespacedHash>> NamespaceProof<M> {
                 .collect();
             tree.inner.check_range_proof(
                 root,
-                &mut &leaf_hashes[..],
+                &leaf_hashes[..],
```
Comment (oblique) [inline src/nmt_proof.rs]: `[..]` can be avoided

File: src/lib.rs (hunk 3 above)
Comment (zvolin) [reply src/lib.rs]: done

File: src/nmt_proof.rs (hunk 4 above)
Comment (zvolin) [reply src/nmt_proof.rs]: done

File: src/lib.rs (hunk 2 above)
Comment (zvolin) [reply src/lib.rs]: done

File: src/lib.rs (hunk 1 above)
Comment (zvolin) [reply src/lib.rs:120]: done

**Pull request merged**